projects/ids-enterprise-ng/src/lib/toolbar-flex/soho-toolbar-flex.component.ts

```diff
@@ -0,0 +1,323 @@
+  }
+
+  private markForRefresh() {
+    this.searchFieldChanged = true;
+
+    // ... make sure the change detector kicks in, otherwise if the inputs
+    // were change programmatially the component may not be eligible for
+    // updating.
+    this.changeDetector.markForCheck();
+  }
+}
+
+/**
+ * soho toolbar flex more button
+ */
+@Component({
+  selector: 'soho-toolbar-flex-more-button', // tslint:disable-line
+  template: `<button class="btn-actions">
+    <svg class="icon" focusable="false" aria-hidden="true" role="presentation">
+      <use xlink:href="#icon-more"></use>
+    </svg>
+    <span class="audible">More Actions</span>
+  </button>
+  <ng-content></ng-content>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SohoToolbarFlexMoreButtonComponent {
+  @HostBinding('class.more') get isMoreButton() { return true; }
+  @HostBinding('class.toolbar-section') get isToolbarSection() { return true; }
+  @Input() isDisabled = false;
+  @Input() ajaxBeforeFunction: Function;
+  @Input() menuId: string;
+}
+
+/**
+ * soho toolbar flex page title.
+ */
+@Component({
+  selector: '[soho-toolbar-flex-page-title]', // tslint:disable-line
+  template: `<ng-content></ng-content>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SohoToolbarFlexPageTitleComponent {
+  @HostBinding('class.page-title') get isPageTitle() { return true; }
+}
+
+/**
+ * soho toolbar flex section title.
+ */
+@Component({
+  selector: '[soho-toolbar-flex-section-title]', // tslint:disable-line
+  template: `<ng-content></ng-content>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SohoToolbarFlexSectionTitleComponent {
+  @HostBinding('class.section-title') get isSectionTitle() { return true; }
+}
+
+/**
+ * Generic soho toolbar flex section
+ */
+@Component({
+  selector: 'soho-toolbar-flex-section', // tslint:disable-line
+  template: `<ng-content></ng-content>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SohoToolbarFlexSectionComponent {
+  @HostBinding('class.toolbar-section') get isToolbarSection() { return true; }
+  @Input() @HostBinding('class.favor') isTitleFavor = false;
+  @Input() @HostBinding('class.title') isTitle = false;
+  @Input() @HostBinding('class.buttonset') isButtonSet = false;
+  @Input() @HostBinding('class.search') isSearch = false;
+}
+
+/**
+ * Soho toolbar Navigation button
+ */
+@Component({
+  selector: 'button[soho-toolbar-flex-nav-button]', // tslint:disable-line
+  template: `
+    <span class="icon app-header">
+        <span class="one"></span>
+        <span class="two"></span>
+        <span class="three"></span>
+    </span>
+    <span class="audible">
+        <ng-content></ng-content>
+    </span>
+            `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SohoToolbarFlexNavButtonComponent {
+  @HostBinding('class.btn-icon') get isIconButton() { return true; }
```

Comment:
I'd like to see static things like this a styles block in the @Component meta

Reply:
I see now, just change to not use a getter. 

`@HostBinding('class.btn-icon') isIconButton = true;`